lib/rust/profiler/data/src/bin/intervals.rs

```diff
@@ -0,0 +1,101 @@
+//! Tool that generates interval reports from profiling data.
+//!
+//! # Usage
+//!
+//! The tool reads a JSON-formatted event log from stdin, and writes a report to stdout.
+//!
+//! For example:
+//!
+//! ```console
+//! ~/git/enso/data $ cargo run --bin intervals < profile.json | less
+//! ```
+
+// === Features ===
+#![feature(test)]
+// === Standard Linter Configuration ===
+#![deny(non_ascii_idents)]
+#![warn(unsafe_code)]
+// === Non-Standard Linter Configuration ===
+#![deny(unconditional_recursion)]
+#![warn(missing_copy_implementations)]
+#![warn(missing_debug_implementations)]
+#![warn(missing_docs)]
+#![warn(trivial_casts)]
+#![warn(trivial_numeric_casts)]
+#![warn(unused_import_braces)]
+
+use enso_profiler_data as data;
+use std::collections;
+
+
+
+// ============
+// === main ===
+// ============
+
+fn main() {
+    use std::io::Read;
+
+    let mut log = String::new();
+    std::io::stdin().read_to_string(&mut log).unwrap();
+    let profile: data::Profile<()> = log.parse().unwrap();
+    let mut aggregator = data::aggregate::Aggregator::default();
+    aggregator.visit_profile(&profile);
+    let root = data::aggregate::Frame::from(aggregator);
+    let funcs = FuncCollector::run(&root);
+    let mut funcs: Vec<_> = funcs.into_iter().collect();
+    funcs.sort_by_key(|(_, func)| 0 - (func.self_duration * 10.0).round() as i64);
```

Comment:
I'd love to see either docs or refactoring to a function/lambda this equation, because it is not clear to me why it works this way.

Reply:
You're right, that's completely opaque. I've rewritten it.